MapboxNavigation/StatusView.swift

```diff
@@ -117,62 +138,106 @@ public class StatusView: UIControl {
         }
     }
     
+    @available(*, deprecated, message: "Add a status using addNewStatus instead")
     public func showStatus(title: String, spinner spin: Bool = false, duration: TimeInterval, animated: Bool = true, interactive: Bool = false) {
-        show(title, showSpinner: spin, interactive: interactive)
+        // show(title, showSpinner: spin, interactive: interactive)
         guard duration < .infinity else { return }
         hide(delay: duration, animated: animated)
     }
     
+    func addNewStatus(status: Status) {
+        guard let firstWord = status.id.components(separatedBy: " ").first else { return }
+        if let row = statuses.firstIndex(where: {$0.id.contains(firstWord)}) {
+            statuses[row] = status
+        } else {
+            statuses.append(status)
+        }
+        manageStatuses()
+    }
+    
+    /**
+     Manages showing and hiding Statuses and the status view.
+     */
+    func manageStatuses(status: Status? = nil) {
+        if statuses.isEmpty {
+            hide(delay: status?.duration ?? 0, animated: status?.animated ?? true)
+        } else {
+            // if we hide a Status and there are Statuses left in the statuses array, show the Status with highest priority
+            guard let highestPriorityStatus = statuses.min(by: {$0.priority.rawValue < $1.priority.rawValue}) else { return }
+            show(status: highestPriorityStatus)
+            hide(with: highestPriorityStatus, delay: highestPriorityStatus.duration)
+        }
+    }
+    
+    /**
+     Hides a given Status without hiding the status view.
+     */
+    func hideStatus(using status: Status?) {
+        guard let firstWord = status?.id.components(separatedBy: " ").first else { return }
+        guard let row = statuses.firstIndex(where: {$0.id.contains(firstWord)}) else { return }
+        let removedStatus = statuses.remove(at: row)
+        manageStatuses(status: removedStatus)
+    }
+    
     func showSimulationStatus(speed: Int) {
         let format = NSLocalizedString("USER_IN_SIMULATION_MODE", bundle: .mapboxNavigation, value: "Simulating Navigation at %@×", comment: "The text of a banner that appears during turn-by-turn navigation when route simulation is enabled.")
         let title = String.localizedStringWithFormat(format, NumberFormatter.localizedString(from: speed as NSNumber, number: .decimal))
-        showStatus(title: title, duration: .infinity, interactive: true)
+        let simulationStatus = Status(id: title, duration: .infinity, interactive: true, priority: StatusView.Priority(rawValue: 2))
+        addNewStatus(status: simulationStatus)
     }
     
     /**
      Shows the status view with an optional spinner.
      */
-    public func show(_ title: String, showSpinner: Bool, interactive: Bool = false) {
```

Comment:
It'd be better to extend existing APIs instead of replacing existing ones. If you feel that newly added API is more concise previous one can be deprecated/or left without any changes.